app_dart/lib/src/service/luci_build_service/build_tags.dart

```diff
@@ -27,11 +27,37 @@ final class BuildTags {
     _buildTags.add(buildTag);
   }
 
+  /// Adds [buildTag] if there is no existing build tag with a matching key.
+  ///
+  /// Returns the build tag that exists at the end of this operation.
+  T addIfAbsent<T extends BuildTag>(T buildTag) {
+    final index = _buildTags.indexWhere((e) => e._key == buildTag._key);
+    if (index == -1) {
+      _buildTags.add(buildTag);
+      return buildTag;
+    }
+    return _buildTags[index] as T;
+  }
+
   /// Adds multiple [buildTags].
   void addAll(Iterable<BuildTag> buildTags) {
     _buildTags.addAll(buildTags);
   }
 
+  /// Replaces an existing build tag with a matching key with [buildTag].
+  ///
+  /// Returns the existing tag that was replaced, or `null` if not replaced.
```

Comment:
Why return the old value at all? Just a boolean for "existed" would be good enough.

Reply:
Done.